edb/schema/scalars.py

```diff
@@ -677,55 +677,48 @@ def apply(
         self.scls = scls
         assert isinstance(scls, ScalarType)
 
-        cur_labels = scls.get_enum_values(schema)
-
-        if cur_labels:
-
+        if self.scls.is_concrete_enum(schema):
```

Comment:
Could we add a comment about the cursed nature of `enum_values` along this code path? This nicely avoids the problem, and is definitely cleaner, but it's still a potential landmind in the future